app/Livewire/Project/Application/Domains.php

```
@@ -0,0 +1,1576 @@
+            foreach ($serviceNames as $serviceName) {
+                // data_get treats dots as path separators; read the service entry by array key.
+                $serviceEntry = $domains[$serviceName] ?? null;
+                $storedRedirect = is_array($serviceEntry) ? ($serviceEntry['redirect'] ?? null) : null;
+                $this->serviceRedirects[$this->serviceRedirectWireKey($serviceName)] = $this->normalizeRedirect(
+                    is_string($storedRedirect) ? $storedRedirect : null
+                );
+            }
+        }
+
+        $this->domainRows = $this->buildDomainRows();
+    }
```

PHPMD (2.15.0)

[warning] The method loadDomainState() has a Cyclomatic Complexity of 19. The configured cyclomatic complexity threshold is 10. (undefined)

(CyclomaticComplexity)